### PR TITLE
[CBRD-20567]Add test cases for 'Allow Expression in Limit Clause'

### PR DESCRIPTION
**Pull request by slamdunkorchid**
http://jira.cubrid.org/browse/CBRD-20567

### CODE REVIEW

**Comment by fanzaiqiang** [comment]
It's better to decrease data size for below test cases:
_004_index_keylimit_big_table.sql
=> The ha_repl test will be affected on performance

expression_01_delete.sql
expression_01_update.sql
expression_02_delete.sql
expression_02_update.sql
index_skip_scan_01_insert.sql
index_skip_scan_01.sql
sort_limit_insert.sql
sort_limit_prepare.sql
sort_limit.sql
sort_limit_update.sql
=> Maybe memory leak test will be affected on performance.  Please decrease level for "from db_root connect by level <=1000". This is not good solution to generate many rows.

filter_index_01_insert.sql
func_index_01_insert.sql
index_skip_scan_01_insert.sql
multi_range_opt_01_insert.sql
=> Mistake to drop 't' instead of correct tables like 'x'.

multi_range_opt_01.sql
multi_range_opt_01_insert.sql
=> Failed on my machine. Please check whether they are unstable.

File: sql/_30_banana_pie_qa/issue_20567_keylimit/cases/update_01.sql [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+CREATE INDEX i_tab_j_k ON tab (j, k);
+INSERT INTO tab VALUES (1,2,3),(6,4,2),(3,4,1),(5,2,1),(1,5,5),(2,6,6),(3,5,4);
+UPDATE STATISTICS on tab;
+update tab set k=918
+WHERE j > 0;
+
+update tab set k=917
+WHERE j > 0
+ using index i_tab_j_k keylimit  0-0-0-0-0+2,10*10;
+select * from tab;
```
Comment (jinmeihua) [inline sql/_30_banana_pie_qa/issue_20567_keylimit/cases/update_01.sql]: Please add order by clause to make result stable.

File: sql/_30_banana_pie_qa/issue_20567_keylimit/cases/multi_range_opt_01_insert.sql [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+insert into x select /*+ recompile */ id,score from student group by s_class having score>=90 order by score desc limit 0*8,0+1;
+select * from x order by 1,2;
+insert into x select /*+ recompile */ id,score from student where s_class in ('1','2') and score>=90 order by score limit 0*5,4+0;
+select * from x order by 1,2;
+insert into x select /*+ recompile */ id,score from student where s_class in ('1','3') and score>=90 order by score desc limit 4-4,2+3;
+select * from x order by 1,2;
+insert into x select /*+ recompile */ id,score from student where score>=90 order by score limit 0*1,4*1;
+select * from x order by 1,2;
+set system parameters 'multi_range_optimization_limit=100';
+drop student;
```
Comment (jinmeihua) [inline sql/_30_banana_pie_qa/issue_20567_keylimit/cases/multi_range_opt_01_insert.sql]: drop x;

File: sql/_30_banana_pie_qa/issue_20567_keylimit/cases/multi_range_opt_01_insert.sql [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+insert into student(s_class,score) values('4',93),('4',93),('4',93),('4',93),('4',93),('4',93),('4',93);
+create index i_s_23 on student(s_class,score);
+update statistics on student;
+set optimization level=513;
+set system parameters 'multi_range_optimization_limit=10000';
+
+drop if exists x;
+create table  x(id int ,score int);
+
+insert into x select /*+ recompile */ id,score from student where s_class in ('1','3') and score>=90 order by score desc limit 0,10*10;
```
Comment (jinmeihua) [inline sql/_30_banana_pie_qa/issue_20567_keylimit/cases/multi_range_opt_01_insert.sql]: Only order by score column, result will not stable, please stabilize test case.

**Comment by fanzaiqiang** [comment]
There are unstable test cases:
One of failure:
```
$ diff filter_index_01_insert.result ../answers/filter_index_01_insert.answer 
19c19
< 4     book4     2010-12-24 00:00:00.0     
---
> 3     book3     2009-12-24 00:00:00.0 
```

**Comment by jinmeihua** [comment]
Based on 76dea12, there are 3 unstable test cases :

- update_01.sql
```
$ diff update_01.result update_01.answer 
128c128
< 1     5     92     
---
> 1     5     95     
131c131
< 3     5     94     
---
> 3     5     92     
```
- group_by.sql
```
$ diff group_by.result group_by.answer 
100c100
< 5     5     
---
> 5     4     
```
- group_by_insert.sql
```
$ diff group_by_insert.result group_by_insert.answer 
504c504
< 5     5     
---
> 5     4     
561c561
< 5     5     
---
> 5     4     
```

File: sql/_30_banana_pie_qa/issue_20567_keylimit/cases/filter_index_01_update.sql [hunk 4]
```diff
@@ -0,0 +1,33 @@
+--filter index
+drop if exists t;
+create table t(col1 int,col2 varchar(20),col3 timestamp);
+insert into t values(1,'book1','2007-12-24');
+insert into t values(2,'book2','2008-12-24');
+insert into t values(3,'book3','2009-12-24');
+insert into t values(4,'book4','2010-12-24');
+insert into t values(2,'book2','2008-12-24');
+create index idx1  on t(col3) where year(col3)>2008;
+select * from t where year(col3)>2008 limit 0,1;
```
Comment (jinmeihua) [inline sql/_30_banana_pie_qa/issue_20567_keylimit/cases/filter_index_01_update.sql]: select * from t where year(col3)>2008 limit 0,1;
->
select * from t where year(col3)>2008 order by 1 limit 0,1;

**Pull request merged**